hres_schema/js/hres_reporting_projects.js

```diff
@@ -0,0 +1,86 @@
+
+P.CanViewResearcherProjectsDashboard = O.action("hres:action:view-researcher-projects-dashboard").
+    title("View Researcher Projects Dashboard").
+    allow("group", Group.Everyone);
+
+// --------------------------------------------------------------------------
+
+P.implementService("std:reporting:discover_collections", function(discover) {
+    discover("projects", "Projects");
+});
+
+P.implementService("std:reporting:collection:projects:setup", function(collection) {
+    collection.
+        currentObjectsOfType(T.Project).
+        fact("principalInvestigator",       "ref",      "Principal Investigator");
+});
+
+P.implementService("std:reporting:gather_collection_update_rules", function(rule) {
+    rule("projects", T.Project, A.Researcher);
+});
+
+P.implementService("std:reporting:collection:projects:get_facts_for_object", 
+    function(object, row) {
+        row.principalInvestigator = object.first(A.Researcher, Q.PrincipalInvestigator);
+    });
+
+// --------------------------------------------------------------------------
+
+P.implementService("std:action_panel:researcher", function(display, builder) {
+    if(O.currentUser.allowed(P.CanViewResearcherProjectsDashboard)) {
+        builder.panel(2000).link("default",
```

Comment:
This will add additional UI into all hres systems, even if they're not using projects. Shouldn't all this functionality be in a separate hres_schema_projects plugin?